cmd/lifecycle/analyzer.go

```diff
@@ -75,6 +78,12 @@ func (a *analyzeCmd) Args(nargs int, args []string) error {
 }
 
 func (a *analyzeCmd) Privileges() error {
+	keychain, err := a.resolveKeychain() // TODO: call resolveKeychain only if needed (useDaemon==false or using a cache image)
```

Comment:
Thinking about this more, I don't think we want to gate calling `a.resolveKeychain()` on this logic as it'd essentially be duplicating the logic in `resolveKeychain()` itself. Maybe things are okay as they are - `ResolveKeychain` will not even try to read the `DOCKER_CONFIG` directory if `CNB_REGISTRY_AUTH` is unset.

Reply:
@natalieparellano, I'm marking this conversation as resolved.
Feel free to reopen it if you think that I missed something.